india_compliance/gst_india/doctype/gst_settings/gst_settings.py

```diff
@@ -101,20 +101,35 @@ def validate_e_invoice_applicability_date(self):
         if not self.enable_api or not self.enable_e_invoice:
             return
 
-        if not self.e_invoice_applicable_from:
+        if (
+            not self.e_invoice_applicable_from
+            and not self.apply_e_invoice_only_for_selected_companies
+        ):
             frappe.throw(
-                _("{0} is mandatory for enabling e-Invoice").format(
-                    frappe.bold(self.meta.get_label("e_invoice_applicable_from"))
+                _("{0} or {1} is mandatory for enabling e-Invoice").format(
```

Comment:
As `apply_e_invoice_only_for_selected_companies` is not checked, we can keep it simple and keep the existing error. It would then be consistent with front-end validation.